src/Input.js

```diff
@@ -245,7 +245,12 @@ const Input = React.createClass({
       ];
     }
 
-    return <FormGroup {...this.props}>{children}</FormGroup>;
+    if (this.props.type === 'submit') {
+      let {bsStyle, ...other} = this.props; /* eslint no-unused-vars: 0 */
```

Comment:
This is a great valid use case to present to the eslint folks. Have you considered submitting an issue with them so in the future we wouldn't need to do this?